### ISSUE TITLE
clang 3.8.0 build fails with jemalloc 4.2.1 with "unknown attribute 'alloc_size' ignored"

### ISSUE THREAD

**Issue by mkevac** [jemalloc/jemalloc#422]
clang 3.8.0 build fails with jemalloc 4.2.1 with "unknown attribute 'alloc_size' ignored"

```
In file included from ../misc/main.c:116:
/home/marko/.../local/jemalloc/include/jemalloc/jemalloc.h:198:46: error: unknown attribute 'alloc_size' ignored [-Werror,-Wunknown-attributes]
    JEMALLOC_CXX_THROW JEMALLOC_ATTR(malloc) JEMALLOC_ALLOC_SIZE(1);
                                             ^
/home/marko/.../local/jemalloc/include/jemalloc/jemalloc.h:152:50: note: expanded from macro 'JEMALLOC_ALLOC_SIZE'
#    define JEMALLOC_ALLOC_SIZE(s) JEMALLOC_ATTR(alloc_size(s))
                                                 ^
/home/marko/.../local/jemalloc/include/jemalloc/jemalloc.h:201:46: error: unknown attribute 'alloc_size' ignored [-Werror,-Wunknown-attributes]
    JEMALLOC_CXX_THROW JEMALLOC_ATTR(malloc) JEMALLOC_ALLOC_SIZE2(1, 2);
                                             ^
/home/marko/.../local/jemalloc/include/jemalloc/jemalloc.h:153:56: note: expanded from macro 'JEMALLOC_ALLOC_SIZE2'
#    define JEMALLOC_ALLOC_SIZE2(s1, s2) JEMALLOC_ATTR(alloc_size(s1, s2))
                                                       ^
/home/marko/...local/jemalloc/include/jemalloc/jemalloc.h:207:5: error: unknown attribute 'alloc_size' ignored [-Werror,-Wunknown-attributes]
    JEMALLOC_ALLOC_SIZE(2);
    ^
/home/marko/.../local/jemalloc/include/jemalloc/jemalloc.h:152:50: note: expanded from macro 'JEMALLOC_ALLOC_SIZE'
#    define JEMALLOC_ALLOC_SIZE(s) JEMALLOC_ATTR(alloc_size(s))
                                                 ^
/home/marko/.../local/jemalloc/include/jemalloc/jemalloc.h:210:24: error: unknown attribute 'alloc_size' ignored [-Werror,-Wunknown-attributes]
    JEMALLOC_CXX_THROW JEMALLOC_ALLOC_SIZE(2);
                       ^
/home/marko/.../local/jemalloc/include/jemalloc/jemalloc.h:152:50: note: expanded from macro 'JEMALLOC_ALLOC_SIZE'
#    define JEMALLOC_ALLOC_SIZE(s) JEMALLOC_ATTR(alloc_size(s))
                                                 ^
/home/marko/.../local/jemalloc/include/jemalloc/jemalloc.h:216:27: error: unknown attribute 'alloc_size' ignored [-Werror,-Wunknown-attributes]
    JEMALLOC_ATTR(malloc) JEMALLOC_ALLOC_SIZE(1);
                          ^
/home/marko/.../local/jemalloc/include/jemalloc/jemalloc.h:152:50: note: expanded from macro 'JEMALLOC_ALLOC_SIZE'
#    define JEMALLOC_ALLOC_SIZE(s) JEMALLOC_ATTR(alloc_size(s))
                                                 ^
/home/marko/.../local/jemalloc/include/jemalloc/jemalloc.h:219:16: error: unknown attribute 'alloc_size' ignored [-Werror,-Wunknown-attributes]
    int flags) JEMALLOC_ALLOC_SIZE(2);
               ^
/home/marko/.../local/jemalloc/include/jemalloc/jemalloc.h:152:50: note: expanded from macro 'JEMALLOC_ALLOC_SIZE'
#    define JEMALLOC_ALLOC_SIZE(s) JEMALLOC_ATTR(alloc_size(s))
                                                 ^
6 errors generated.

```

**Comment by jasone** [comment]
When I configure 4.2.1 on an Ubuntu 16.04 system, which has clang 3.8.0-2ubuntu4 installed, here's what I see:

  ```
...
checking whether compiler supports -Werror... yes
checking whether alloc_size attribute is compilable... no
...
  ```

Furthermore, after fixing ```EXTRA_CFLAGS``` handling to allow:

  ```EXTRA_CFLAGS="-Werror -Wunknown-attributes" CC=clang ./autogen.sh```

I see no compilation issues, indicating that indeed the ```alloc_size``` attribute is being omitted.  Are you perhaps configuring with gcc, then building with clang?  When I hack the generated ```Makefile``` to try that, I can reproduce the errors, as well as some regarding unsupported ```format``` attributes.

I'm going to close this under the assumption that some sort of operator error (probably involving manual hacking of ```Makefile```).  If I missed something and you still think there is a bug, please reopen with additional information.

**Comment by jasone** [comment]
```EXTRA_CFLAGS``` fix: af0e28fd942d6f3b6198aebdeea6043b3542d096
